Map sidesets for changing mesh using mask (#30388)
idaholab/moose: framework/doc/content/source/meshmodifiers/SidesetAroundSubdomainUpdater.md

## Description

> The `SidesetAroundSubdomainUpdater` mesh modifier updates a sideset during a simulation by reassigning it based on a pair of subdomain sets, the [!param](/MeshModifiers/SidesetAroundSubdomainUpdater/inner_subdomains) and [!param](/MeshModifiers/SidesetAroundSubdomainUpdater/outer_subdomains). The sideset will be updated to comprise all sides along `inner_subdomains` that are neighboring `outer_subdomains` or that have no neighbor. This dynamic update can be useful in conjunction with subdomain update user objects such as [CoupledVarThresholdElementSubdomainModifier](CoupledVarThresholdElementSubdomainModifier.md).  The element sides on which this object acts can be limited using the [!param](/MeshModifiers/SidesetAroundSubdomainUpdater/mask_side) parameter, as shown in:

Reviewer: let s explain this mask more. 

The mask is restricting the sides which can be added to the moving side as the boundary moves.
OR
The mask is restricting the sides which can be removed from the moving side as the boundary moves.

Reviewer: I think it s the former

!listing test/tests/meshmodifiers/sideset_around_subdomain_updater/simple.i block=MeshModifiers

!alert note

> Use the [!param](/MeshModifiers/SidesetAroundSubdomainUpdater/execution_order_group) parameter to ensure that this user object is run after any subdomain changing user object has fully traversed the mesh.  Nodesets are also updated as well as sidesets, so the use of `construct_node_list_from_side_list = false` will not run as expected.

Reviewer: ```suggestion
Use the [!param](/MeshModifiers/SidesetAroundSubdomainUpdater/execution_order_group) parameter to ensure that this user object is run after any subdomain changing user object has fully traversed the mesh. 

!alert warning
Nodesets are also updated as well as sidesets, so the use of `construct_node_list_from_side_list = false` will not run as expected.
```

## Description

> The `SidesetAroundSubdomainUpdater` mesh modifier updates a sideset during a simulation by reassigning it based on a pair of subdomain sets, the [!param](/MeshModifiers/SidesetAroundSubdomainUpdater/inner_subdomains) and [!param](/MeshModifiers/SidesetAroundSubdomainUpdater/outer_subdomains). The sideset will be updated to comprise all sides along `inner_subdomains` that are neighboring `outer_subdomains` or that have no neighbor. This dynamic update can be useful in conjunction with subdomain update user objects such as [CoupledVarThresholdElementSubdomainModifier](CoupledVarThresholdElementSubdomainModifier.md).  The mask is restricting the sides which can be added to the moving side as the boundary moves, [!param](/MeshModifiers/SidesetAroundSubdomainUpdater/mask_side), as shown in:

Reviewer: ```suggestion
The `SidesetAroundSubdomainUpdater` mesh modifier updates a sideset during a simulation by reassigning it based on a pair of subdomain sets, the [!param](/MeshModifiers/SidesetAroundSubdomainUpdater/inner_subdomains) and [!param](/MeshModifiers/SidesetAroundSubdomainUpdater/outer_subdomains). The sideset will be updated to comprise all sides along `inner_subdomains` that are neighboring `outer_subdomains` or that have no neighbor. This dynamic update can be useful in conjunction with subdomain update user objects such as [CoupledVarThresholdElementSubdomainModifier](CoupledVarThresholdElementSubdomainModifier.md).  

 An optional 'mask' can be used to restrict the sides which can be added to the moving side as the boundary moves, [!param](/MeshModifiers/SidesetAroundSubdomainUpdater/mask_side), as shown in:
```

!listing test/tests/meshmodifiers/sideset_around_subdomain_updater/simple.i block=MeshModifiers

!alert note
Use the [!param](/MeshModifiers/SidesetAroundSubdomainUpdater/execution_order_group) parameter to ensure that this user object is run after any subdomain changing user object has fully traversed the mesh.

!alert warning

> Nodesets are also updated as well as sidesets, so the use of `construct_node_list_from_side_list = false` will not run as expected.

Reviewer: ```suggestion
Nodesets are also updated as well as sidesets, so the [!param](/Mesh/FileMesh/construct_node_list_from_side_list) `Mesh` parameter will not be respected.
```

## Description

> The `SidesetAroundSubdomainUpdater` mesh modifier updates a sideset during a simulation by reassigning it based on a pair of subdomain sets, the [!param](/MeshModifiers/SidesetAroundSubdomainUpdater/inner_subdomains) and [!param](/MeshModifiers/SidesetAroundSubdomainUpdater/outer_subdomains). The sideset will be updated to comprise all sides along `inner_subdomains` that are neighboring `outer_subdomains` or that have no neighbor. This dynamic update can be useful in conjunction with subdomain update user objects such as [CoupledVarThresholdElementSubdomainModifier](CoupledVarThresholdElementSubdomainModifier.md).  The ability to remap sidesets as the mesh evolves is also possible through the use of the [!param](/MeshModifiers/SidesetAroundSubdomainUpdate/mask_side).  This is shown in:

Reviewer 2: ```suggestion
The `SidesetAroundSubdomainUpdater` mesh modifier updates a sideset during a simulation by reassigning it based on a pair of subdomain sets, the [!param](/MeshModifiers/SidesetAroundSubdomainUpdater/inner_subdomains) and [!param](/MeshModifiers/SidesetAroundSubdomainUpdater/outer_subdomains). The sideset will be updated to comprise all sides along `inner_subdomains` that are neighboring `outer_subdomains` or that have no neighbor. This dynamic update can be useful in conjunction with subdomain update user objects such as [CoupledVarThresholdElementSubdomainModifier](CoupledVarThresholdElementSubdomainModifier.md).  The element sides on which this object acts can be limited using the [!param](/MeshModifiers/SidesetAroundSubdomainUpdate/mask_side) parameter, as shown in:
```

> !listing test/tests/mesh/add_sideset_ids/simple.i
> block=MeshModifiers

Reviewer: ```suggestion
!listing test/tests/mesh/add_sideset_ids/simple.i block=MeshModifiers
```
